File: informatika ismeretek/emelt/2019_may/java/Keres.java
public class Keres {
    
    public final String cim;
    public final String datumIdo;
    public final String keres;
    public final String httpKod;
    public final String meret;
    
    public Keres(String line) {
        var firstSplit = line.split("\\*");
        var utolsoSzokozIndex = firstSplit[3].indexOf(' ');
        
        cim = firstSplit[0];
        datumIdo = firstSplit[1];
        keres = firstSplit[2];
        httpKod = firstSplit[3].substring(0, utolsoSzokozIndex);
        meret = firstSplit[3].substring(utolsoSzokozIndex + 1);
    }
    
    public int byteMeret() {
        return meret.equals("-") ? 0 : Integer.parseInt(meret);
    }
    
    public boolean domain() {
        return Character.isLetter(cim.charAt(cim.length() - 1));
    }
}

File: informatika ismeretek/emelt/2019_may/c#/Keres.cs
using System;

namespace Erettsegi {
    public struct Keres {
        public readonly string cim;
        public readonly string datumIdo;
        public readonly string keres;
        public readonly string httpKod;
        public readonly string meret;
    
        public Keres(String line) {
            var firstSplit = line.Split('*');
            var utolsoSzokozIndex = firstSplit[3].IndexOf(' ');
        
            cim = firstSplit[0];
            datumIdo = firstSplit[1];
            keres = firstSplit[2];
            httpKod = firstSplit[3].Substring(0, utolsoSzokozIndex);
            meret = firstSplit[3].Substring(utolsoSzokozIndex + 1);
        }
    
        public int ByteMeret() {
            return meret == "-" ? 0 : int.Parse(meret);
        }
    
        public bool Domain() {
            return char.IsLetter(cim[cim.Length - 1]);
        }
    }
}